pandas/core/generic.py

```diff
@@ -8564,6 +8523,246 @@ def nanptp(values, axis=0, skipna=True):
             ``numpy.ndarray`` method ``ptp``.""",
             nanptp)
 
+    _shared_docs['transform'] = ("""
+    Apply a function that returns a same-shaped object.
+
+    .. versionadded:: 0.20.0
+
+    Parameters
+    ----------
+    func : callable, list of callables, or dict
+        The function or function to apply to the DataFrame or Series.
+        Each function should expect a Series and return an array-like
+        with the same shape. Accepted combinations are:
+
+        - function : Applied to the Series, or to each column of the
+          DataFrame independently.
+        - list of functions : Each function is applied to the Series
+          or to each column of the DataFrame independently.
+        - dict of column names -> functions or list of functions :
+          For DataFrames only, apply the function or functions for a
+          given key to a specific column.
+
+    axis : {0 or 'index', 1 or 'columns'}, default 'index'
```

Comment:
@jorisvandenbossche @datapythonista what to do here, do you think? `axis` isn't a keyword of `transform`, but it's intercepted by the implementation, and not passed through to `func`. I've chosen to document it. I think the best thing to do is to include it in the function signature as well (though in a separate PR).

Reply:
Sorry, didn't see that during the sprint. As the sprint is over, and this PR changes code, I think this could be addressed here, right?